### ISSUE TITLE
2.3.11 breaks isLinkCurrent and {ifCurrent} macro

### ISSUE THREAD

**Issue by ondrejmirtes** [nette/application#124]
Following-up on my [bug report](https://github.com/nette/application/pull/112) about typehinted scalar parameters. The fix that resolved this bug also introduced a rather inconvenient BC break.

With non-typehinted scalar parameters, it's not necessary to pass all parameters to isLinkCurrent method:

`$presenter->isLinkCurrent(':Homepage:default', []);`

In this example, calling the method works even if the requested action has required arguments. By calling it without the arguments, I simply mean to ask "Is the current page at Homepage:default?" without checking exact parameters.

If the action has PHP-7-scalar typehints, I cannot longer call isLinkCurrent without its arguments:

```php
public function renderDefault(int $id)
{
}
...
$presenter->isLinkCurrent(':Homepage:default', []); // fails on PHP-7
```

BC break aside, I think it also reduces the scope where isLinkCurrent is usable. Before PHP 7, I could use it for checking whether am I in a concrete Presenter:action pair without caring about exact arguments.

For example if I want to always render a template block when I am at `Article:default`, in the past I could have used `{ifCurrent Article:default}`. Now, I have to pass the arguments to the macro.

Ideally, the isLinkCurrent/ifCurrent functionality should not depend on generating links because it should behave independently of each other because of different requirements for these functionalities.

If this issue is a logical conclusion of the suggested behaviour in #112 and the current behaviour is fine for everyone, feel free to close this issue.

Thanks!

**Comment by dg** [comment]
Good point, can you send PR?

**Comment by ondrejmirtes** [comment]
I opened #125 which tests current behaviour of isLinkCurrent - I found out there are no tests for this method so I added several test cases and refactored tests to be more compliant with DRY.

I'd like if you'd verified that the tests are OK because they will be a foundation for the isLinkCurrent rewrite.

I also opened #126 which now contains failing tests for PHP 7. The asserts for isLinkCurrent are now reused across PHP < 7 and >= 7 versions so the difference in behaviour is quite clear.
